### ISSUE TITLE
Get rid of https://demo.productionready.io/main.css to force proper frontend asset bundle integration

### ISSUE THREAD

**Issue by cirosantilli** [gothinkster/realworld#662]
CSS asset bundling is a fundamental part of frontend development, but currently we cheat it out by just using https://demo.productionready.io/main.css

Instead, frontends should use https://github.com/gothinkster/conduit-bootstrap-template either via:

- a npm package (could be direct install from git for scss-capable frontends)
- a git submodule

to demonstrate how to integrate the (hopefully .scss) entry into their asset bundler.

Also note that https://demo.productionready.io/main.css and https://github.com/gothinkster/conduit-bootstrap-template are out of sync currently, building from source gives a broken CSS: https://github.com/gothinkster/conduit-bootstrap-template/issues/5

Unfortunately there's no easy way to enforce this besides removing https://demo.productionready.io/main.css entirely. On the short term unmaintained frontends would break. But on the long term this is the correct path for the project. Maybe v2 will break enough things that it won't matter. Not sure.

**Comment by geromegrignon** [comment]
Hello @cirosantilli ,

> Maybe v2 will break enough things that it won't matter. Not sure.

Indeed, we plan to introduce a new brand ui design for the V2 (part of the explanation being to avoid confusion with the first version and related changes).

Using a git submodule is a great idea : using npm would require an account and one of our current points of focus is to avoid reliying on external accounts for a sustainable maintenance.

About the [conduit-bootrap-template](https://github.com/gothinkster/conduit-bootstrap-template), we won't fix it for the current version as it works.

**Comment by cirosantilli** [comment]
Ah fantastic, if there's going to be a new CSS, that makes things simpler.

**Comment by plusgut** [comment]
I think it is very problematic to enforce that all the css sits in one file.
With that approach technology which uses webcomponents will have a bin problem, since css does not penetrate these components.
Also this hinders the possibility to implement in an idiomatic way of any given environment.

To give one full fledged css, would indeed be the easy way of letting people start implementing, but I think it would be beneficial if there is a way to get  the css needed for each little component.

**Comment by mvila** [comment]
I agree that the v2 specs should not enforce the usage of a specific CSS file. A CSS file could be provided as an example, but then it'd be up to the implementor to style its frontend using the most appropriate technique given the framework/libraries he's using.

**Comment by plusgut** [comment]
I like that idea a lot.
Further I think it would be helpful if the css is tiny and does not include a big css library. I do not believe that in a modern website there is a need for that anyway, since css-grid and custom-properties exists nowadays.

**Comment by cirosantilli** [comment]
I also didn't mean to propose that we should enforce how implementations implement their CSS.

My main recommendation is that we don't host a reference CSS file where it can be easily included, to make it more likely that implementations will actually bundle it properly. Just put a SASS reference in this repo and be done with it.

Implementations would then be allowed to re-implement it idiomatically (which implies proper bundling), though in practice most frontends will likely just bundle what we provide by default to avoid the reimplementation effort.

**Comment by geromegrignon** [comment]
> I like that idea a lot.
> Further I think it would be helpful if the css is tiny and does not include a big css library. I do not believe that in a modern website there is a need for that anyway, since css-grid and custom-properties exists nowadays.

I agree. I would like to get rid of bootstrap with the new version.

**Comment by plusgut** [comment]
> > I like that idea a lot.
> > Further I think it would be helpful if the css is tiny and does not include a big css library. I do not believe that in a modern website there is a need for that anyway, since css-grid and custom-properties exists nowadays.
> 
> I agree. I would like to get rid of bootstrap with the new version.

That makes me so happy to hear that from you!
Is there any way I can help out at V2?

**Comment by geromegrignon** [comment]
Hi @plusgut, i'll let you know when we'll actively work on V2.

**Comment by geromegrignon** [comment]
Hi @plusgut, it's been a while. I'll create a repository to work on the V2 ui design in a few days.

I'll integrate storybook as a tool to demonstrate small parts of the application in isolation and facilitates discussion about future changes.

The goal of this step towards v2 is to work on a framework agnostic version (pure HTML/CSS).
It'll replace the current 'template' part of the current documentation.

I'll also create another dedicated repository for the V1, demonstrating how storybook works with current stylesheet and templates.

For your information, this new repository will be tagged with 'hacktoberfest' in case you plan to participate.

**Comment by plusgut** [comment]
@geromegrignon that sounds great, I'm looking forward to it. but I probably won't have much time available to participate.

**Comment by geromegrignon** [comment]
@plusgut here is a quite simple starter example using the existing templates: https://realworld-storybook-old.netlify.app/?path=/story/pages-article-form--article-form
I just created the repo for the next version: https://github.com/gothinkster/realworld-storybook
